### PR TITLE
[TACHYON-1281] Remove SuspectedFileSizeException

### PR DESCRIPTION
**Pull request by ShaoshanLiu**
use more specific exceptions for different cases

### CODE REVIEW

**Comment by ShaoshanLiu** [comment]
sorry, put the wrong title here..let me recreate a new one

File: common/src/main/java/tachyon/exception/FileAlreadyCompletesException.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.exception;
+
+public class FileAlreadyCompletesException extends TachyonException {
```
Comment (jsimsa) [inline common/src/main/java/tachyon/exception/FileAlreadyCompletesException.java]: s/FileAlreadyCompletes/FileAlreadyCompleted/

File: common/src/main/java/tachyon/exception/InvalidFileSizeException.java [hunk 2]
```diff
@@ -15,25 +15,26 @@
 
 package tachyon.exception;
 
-public class SuspectedFileSizeException extends TachyonException {
+// this exception shows the file size is invalid
```
Comment (jsimsa) [inline common/src/main/java/tachyon/exception/InvalidFileSizeException.java]: Use JavaDoc style. So:

```
/**
 * This exception indicates the file size is invalid.
 */
```

File: common/src/main/java/tachyon/exception/TachyonExceptionType.java [hunk 3]
```diff
@@ -24,12 +24,13 @@
...
   FILE_ALREADY_EXISTS(FileAlreadyExistsException.class),
   FILE_DOES_NOT_EXIST(FileDoesNotExistException.class),
+  FILE_ALREADY_COMPLETES(FileAlreadyCompletesException.class),
   INVALID_PATH(InvalidPathException.class),
   INVALID_WORKER_STATE(InvalidWorkerStateException.class),
   LINEAGE_DELETION(LineageDeletionException.class),
   LINEAGE_DOES_NOT_EXIST(LineageDoesNotExistException.class),
   NO_WORKER(NoWorkerException.class),
-  SUSPECTED_FILE_SIZE(SuspectedFileSizeException.class),
+  SUSPECTED_FILE_SIZE(InvalidFileSizeException.class),
```
Comment (jsimsa) [inline common/src/main/java/tachyon/exception/TachyonExceptionType.java]: s/SUSPECTED_FILE_SIZE/INVALID_FILE_SIZE/

File: servers/src/main/java/tachyon/master/file/FileSystemMaster.java [hunk 4]
```diff
@@ -248,12 +249,14 @@ public void stop() throws IOException {
    * @param fileId the file id
    * @param length the length of the file
    * @return true on success
-   * @throws SuspectedFileSizeException
+   * @throws InvalidFileSizeException
    * @throws BlockInfoException
    * @throws FileDoesNotExistException
+   * @throws FileAlreadyCompletesException if setting length on a file that already completes
```
Comment (jsimsa) [inline servers/src/main/java/tachyon/master/file/FileSystemMaster.java]: s/that already completes/that is already completed/

File: servers/src/main/java/tachyon/master/file/FileSystemMaster.java [hunk 5]
```diff
@@ -248,12 +249,14 @@ public void stop() throws IOException {
    * @param fileId the file id
    * @param length the length of the file
    * @return true on success
-   * @throws SuspectedFileSizeException
+   * @throws InvalidFileSizeException
```
Comment (jsimsa) [inline servers/src/main/java/tachyon/master/file/FileSystemMaster.java]: it would be great if you could add missing exception descriptions for this method JavaDoc

**Comment by jsimsa** [comment]
@ShaoshanLiu thanks for you for PR, left some comments for you

File: common/src/main/java/tachyon/exception/FileAlreadyCompletesException.java (hunk 1 above)
Comment (ShaoshanLiu) [inline common/src/main/java/tachyon/exception/FileAlreadyCompletesException.java]: Fixed

File: common/src/main/java/tachyon/exception/InvalidFileSizeException.java (hunk 2 above)
Comment (ShaoshanLiu) [inline common/src/main/java/tachyon/exception/InvalidFileSizeException.java]: Fixed

File: common/src/main/java/tachyon/exception/TachyonExceptionType.java (hunk 3 above)
Comment (ShaoshanLiu) [inline common/src/main/java/tachyon/exception/TachyonExceptionType.java]: fixed

**Comment by calvinjia** [comment]
@ShaoshanLiu I was hoping we could remove the exception completely and move the filesize exception into an existing exception type. Are there any good fits?

File: servers/src/main/java/tachyon/master/file/FileSystemMaster.java (hunk 4 above)
Comment (ShaoshanLiu) [inline servers/src/main/java/tachyon/master/file/FileSystemMaster.java]: fixed

File: servers/src/main/java/tachyon/master/file/FileSystemMaster.java (hunk 5 above)
Comment (ShaoshanLiu) [inline servers/src/main/java/tachyon/master/file/FileSystemMaster.java]: fixed

**Comment by ShaoshanLiu** [comment]
@calvinjia hard to find one that fits, so I added one to make it more specific.  thanks!

**Comment by calvinjia** [comment]
@ShaoshanLiu Maybe just `IllegalArgument`/`State` exceptions instead of the new ones?

**Comment by ShaoshanLiu** [comment]
@calvinjia looking at other exceptions within Tachyon, they are very specific, for example, FILE_ALREADY_EXISTS, FILE_DOES_NOT_EXISTS, do you want to change the exceptions to become more generic?  Thanks.

**Comment by jsimsa** [comment]
LGTM

File: common/src/main/java/tachyon/exception/TachyonExceptionType.java [hunk 6]
```diff
@@ -24,12 +24,13 @@
   FAILED_TO_CHECKPOINT(FailedToCheckpointException.class),
   FILE_ALREADY_EXISTS(FileAlreadyExistsException.class),
   FILE_DOES_NOT_EXIST(FileDoesNotExistException.class),
+  FILE_ALREADY_COMPLETES(FileAlreadyCompletedException.class),
```
Comment (gpang) [inline common/src/main/java/tachyon/exception/TachyonExceptionType.java]: FILE_ALREADY_COMPLETES -> FILE_ALREADY_COMPLETED

Also, sort it alphabetically?

File: common/src/main/java/tachyon/exception/TachyonExceptionType.java [hunk 7]
```diff
@@ -24,12 +24,13 @@
...
   INVALID_PATH(InvalidPathException.class),
   INVALID_WORKER_STATE(InvalidWorkerStateException.class),
   LINEAGE_DELETION(LineageDeletionException.class),
   LINEAGE_DOES_NOT_EXIST(LineageDoesNotExistException.class),
   NO_WORKER(NoWorkerException.class),
-  SUSPECTED_FILE_SIZE(SuspectedFileSizeException.class),
+  INVALID_FILE_SIZE(InvalidFileSizeException.class),
```
Comment (gpang) [inline common/src/main/java/tachyon/exception/TachyonExceptionType.java]: sort it alphabetically?

**Comment by gpang** [comment]
I just left a few minor comments. Thanks @ShaoshanLiu !

**Comment by ShaoshanLiu** [comment]
@gpang  fixed, thanks!

**Comment by yupeng9** [comment]
LGTM

**Comment by ShaoshanLiu** [comment]
would someone please merge if not other comments? Thanks!

**Comment by yupeng9** [comment]
@ShaoshanLiu , could you resolve the conflicts? It should be good for merge afterwards.

File: servers/src/main/java/tachyon/master/file/meta/InodeFile.java [hunk 8]
```diff
@@ -254,15 +255,15 @@ public synchronized void setCompleted(long length) {
    * negative.
    *
    * @param length The new length of the file, cannot be negative
-   * @throws SuspectedFileSizeException
+   * @throws InvalidFileSizeException
```
Comment (calvinjia) [inline servers/src/main/java/tachyon/master/file/meta/InodeFile.java]: Could you complete the javadoc here?

**Comment by calvinjia** [comment]
@ShaoshanLiu Thanks for the patch, looks good other than a minor javadoc issue.

File: common/src/main/java/tachyon/exception/InvalidFileSizeException.java [hunk 9]
```diff
@@ -15,26 +15,26 @@
 
 package tachyon.exception;
 
-// TODO(jiri): Rename this to InvalidFileLengthException
-public class SuspectedFileSizeException extends TachyonException {
+// this exception indicates the file size is invalid
```
Comment (jsimsa) [inline common/src/main/java/tachyon/exception/InvalidFileSizeException.java]: Can this be a proper JavaDoc?

File: servers/src/test/java/tachyon/master/file/meta/InodeFileTest.java [hunk 10]
```diff
@@ -58,19 +59,17 @@ public void setLengthTest() throws Exception {
...
   }
 
   @Test
-  public void completeTwiceTest() throws Exception {
-    mThrown.expect(SuspectedFileSizeException.class);
-    mThrown.expectMessage("File testFile1 has already been completed.");
-
+  public void setLengthAfterCompleteTest() throws Exception {
+    mThrown.expect(FileAlreadyCompletedException.class);
+    mThrown.expectMessage("InodeFile has been completed.");
```
Comment (jsimsa) [inline servers/src/test/java/tachyon/master/file/meta/InodeFileTest.java]: undo this change

File: common/src/main/java/tachyon/exception/InvalidFileSizeException.java (hunk 9 above)
Comment (ShaoshanLiu) [inline common/src/main/java/tachyon/exception/InvalidFileSizeException.java]: I will remove this line to follow the convention of other exception classes

File: servers/src/test/java/tachyon/master/file/meta/InodeFileTest.java (hunk 10 above)
Comment (ShaoshanLiu) [inline servers/src/test/java/tachyon/master/file/meta/InodeFileTest.java]: fixed

File: common/src/main/java/tachyon/exception/InvalidFileSizeException.java (hunk 9 above)
Comment (ShaoshanLiu) [inline common/src/main/java/tachyon/exception/InvalidFileSizeException.java]: fixed

**Comment by ShaoshanLiu** [comment]
timeout building the patch? can we restart?

**Comment by jsimsa** [comment]
Jenkins, test this please.

**Comment by jsimsa** [comment]
Jenkins, test this please.
